Mathlib/Data/ENat/Lattice.lean

```diff
@@ -35,7 +35,9 @@ lemma iInf_coe_ne_top : ⨅ i, (f i : ℕ∞) ≠ ⊤ ↔ Nonempty ι := by
   rw [Ne.def, iInf_coe_eq_top, not_isEmpty_iff]
 lemma iInf_coe_lt_top : ⨅ i, (f i : ℕ∞) < ⊤ ↔ Nonempty ι := WithTop.iInf_coe_lt_top
 
+lemma coe_sSup' : BddAbove s → (↑(sSup s) : ℕ∞) = sSup ((↑) '' s) := WithTop.coe_sSup'
 lemma coe_sSup : BddAbove s → ↑(sSup s) = ⨆ a ∈ s, (a : ℕ∞) := WithTop.coe_sSup
+lemma coe_sInf' : s.Nonempty → (↑(sInf s) : ℕ∞) = sInf ((↑) '' s) := WithTop.coe_sInf'
```

Comment:
I would have hoped you could have done without `coe_sSup'`, `coe_sInf'` because I consider those lemmas to have have noncanonical RHS. Do you really need them? If so, how?

Reply:
If I remember correctly, the issue is that `sSup` can do what `⨆ a ∈ s, ...` takes nested sups to do (and, respectively, infs). I wasn't able to pass through the nested version to get what I needed.

These appear to truly be missing bits of API from `ENat`. If you have concerns about canonicity, you could direct that at WithTop's API, and ENat could follow suit.

Reply:
I guess after the last refinements these were not necessary, so I deleted them.